doc/specs/stdlib_hash_functions.md

```diff
@@ -0,0 +1,1744 @@
+
+##### Class
+
+Subroutine
+
+##### Argument
+
+`seed`: shall be a defined integer scalar variable of kind `INT32`.
+It is an `intent(inout)` argument. On input `seed` should be defined,
+and on output it will be different from the input `seed`.
+
+##### Note
+
+Currently there are no known bad seeds for `PENGY_HASH`, but if any are
+identified the procedure will be revised so that they cannot be
+returned.  This subroutine uses Fortran's intrinsic
+ `RANDOM_NUMBER` and the values returned can be changed by calling the
+ intrinsic `RANDOM_INIT`.
+
+##### Example
+
+See the example for `PENGY_HASH`.
+
+
+#### `NEW_SPOOKY_HASH_SEED`- returns a valid input seed for `SPOOKY_HASH`
+
+##### Status
+
+Experimental
+
+##### Description
+
+Calculates a 32 bit two element vector of "random" integer values that
+is believed to be a valid seed for `SPOOKY_HASH` and is also different
+from the input seed. 
+
+##### Syntax
+
+`code = call [[stdlib_32_bit_hash_codes:new_spooky_hash_seed]]( seed )`
+
+##### Class
+
+Subroutine
+
+##### Argument
+
+`seed`: shall be a defined two element integer vector variable of kind
+`INT32`. It is an `intent(inout)` argument. On input `seed` should be
+defined, and on output it will be different from the input `seed`.
+
+##### Note
+
+Currently there are no known bad seeds for `SPOOKY_HASH`, but if any are
+identified the procedure will be revised so that they cannot be
+returned.  This subroutine uses Fortran's intrinsic
+ `RANDOM_NUMBER` and the values returned can be changed by calling the
+ intrinsic `RANDOM_INIT`.
+
+##### Example
+
+See the example for `SPOOKY_HASH`.
+
+
+#### `ODD_RANDOM_INTEGER` - returns odd integer
+
+##### Status
+
+Experimental
+
+##### Description
+
+Returns a random 64 bit integer distributed uniformly over the odd values.
+
+##### Syntax
+
+`call [[stdlib_64_bit_hash_codes:odd_random_integer]]( harvest )`
+
+##### Class
+
+Subroutine
+
+##### Argument
+
+`harvest`: Shall be an integer of kind `INT64`. It is an `intent(out)`
+argument.
+
+##### Note
+
+`ODD_RANDOM_INTEGER` is intended to generate seeds for
+ `UNIVERSAL_MULT_HASH`. `ODD_RANDOM_NUMBER` uses Fortran's intrinsic
+ `RANDOM_NUMBER` and the values returned can be changed by calling the
+ intrinsic `RANDOM_INIT`.
+
+##### Example
+
+See `UNIVERSAL_MULT_HASH`.
+
+
+#### `PENGY_HASH` - maps a character string or integer vector to an integer
+
+##### Status
+
+Experimental
+
+##### Description
+
+Maps a character string or integer vector to a 64 bit integer whose
+value also depends on a scalar 32 bit integer, `seed`.
+
+##### Syntax
+
+`code = [[stdlib_64_bit_hash_codes:pengy_hash]]( key, seed )`
+
+#####  Class
+
+Pure function
+
+##### Arguments
+
+`key`: shall be a scalar  expression of type default character or a
+Rank 1 integer vector expression of kind `INt8`, `INT16`, `INT32`, or
+`INTT64`. It is an `intent(in)` argument.
+
+`seed`: shall be an integer ex of kind `INT64`. It ispression
```

Comment:
```suggestion
`seed`: shall be an integer ex of kind `INT64`. It is
```

Reply:
Changed to

`seed`: shall be an integer expression of kind `INT64`. It is
an `intent(in)` argument.